src/core/meshcop/joiner.cpp

```diff
@@ -107,6 +107,7 @@ void Joiner::HandleDiscoverResult(otActiveScanResult *aResult)
 {
     if (aResult != NULL)
     {
+        mNetif.GetMac().SetPanId(aResult->mPanId);
```

Comment:
Shouldn't the PAN ID be set after the discover process is complete (e.g. in the `else` block below, as is done with the channel?  Eventually, this `if` block will contain code to prioritize among multiple discover responses.